### ISSUE TITLE
Wrong behavior after using "limit" in CSV.File when the number of limit is larger than the total rows

### ISSUE THREAD

**Issue by bearxu83** [JuliaData/CSV.jl#386]
```
using CSV
#suppose we have a file
text1 = IOBuffer("fullVisitorId,PredictedLogRevenue\n18966949534117,0\n39738481224681,0\n")

# the schema is like
'''
Tables.Schema:
 :fullVisitorId        Union{Missing, Int64}
 :PredictedLogRevenue  Union{Missing, Int64})
'''
println([row.fullVisitorId for row in CSV.File(seek(text1,0))])
#Print the right result
#[18966949534117, 39738481224681]

println([row.fullVisitorId for row in CSV.File(seek(text1,0);limit=3)])
#the total datarows should be 2. Now the limit is 3
#sometimes the result is:
#Union{Missing, Int64}[18966949534117, 39738481224681, missing]
#sometimes the result is:
#["18966949534117", "39738481224681", "fullVisitorId"]
```

Maybe we don't need to add missing row when the limit is beyond the total datarows.

Tested in Windows 10, Julia 1.1

**Comment by Djoop** [comment]
Can be worse than that:

```julia
julia> using CSV
julia> using DataFrames
julia> DataFrame(col1 = [1.0], col2 = ["hi"]) |> CSV.write("test.csv")
"test.csv"
julia> CSV.read("test.csv")
1×2 DataFrame
│ Row │ col1     │ col2    │
│     │ Float64⍰ │ String⍰ │
├─────┼──────────┼─────────┤
│ 1   │ 1.0      │ hi      │
julia> CSV.read("test.csv", limit=3)

signal (11): Erreur de segmentation
in expression starting at no file:0
peekbyte at ./array.jl:729 [inlined]
#parse!#23 at /home/USERNAME/.julia/packages/Parsers/tRRFt/src/Parsers.jl:501 [inlined]
parse! at /home/USERNAME/.julia/packages/Parsers/tRRFt/src/Parsers.jl:499 [inlined]
#parse!#22 at /home/USERNAME/.julia/packages/Parsers/tRRFt/src/Parsers.jl:400 [inlined]
parse! at /home/USERNAME/.julia/packages/Parsers/tRRFt/src/Parsers.jl:400 [inlined]
#parse#11 at /home/USERNAME/.julia/packages/Parsers/tRRFt/src/Parsers.jl:305 [inlined]
parse at /home/USERNAME/.julia/packages/Parsers/tRRFt/src/Parsers.jl:305 [inlined]
trytype at /home/USERNAME/.julia/packages/CSV/eWuJV/src/typedetection.jl:149 [inlined]
detecttype at /home/USERNAME/.julia/packages/CSV/eWuJV/src/typedetection.jl:170
detect at /home/USERNAME/.julia/packages/CSV/eWuJV/src/typedetection.jl:110
unknown function (ip: 0x7f257a1ef22c)
jl_fptr_trampoline at /buildworker/worker/package_linux64/build/src/gf.c:1864
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2219
#File#1 at /home/USERNAME/.julia/packages/CSV/eWuJV/src/CSV.jl:184
unknown function (ip: 0x7f257a1df436)
jl_fptr_trampoline at /buildworker/worker/package_linux64/build/src/gf.c:1864
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2219
Type at ./none:0 [inlined]
#read#105 at /home/USERNAME/.julia/packages/CSV/eWuJV/src/CSV.jl:310
unknown function (ip: 0x7f257a1ddcfb)
jl_fptr_trampoline at /buildworker/worker/package_linux64/build/src/gf.c:1864
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2219
#read at ./none:0 [inlined]
#read at ./none:0
unknown function (ip: 0x7f257a1dd726)
jl_fptr_trampoline at /buildworker/worker/package_linux64/build/src/gf.c:1864
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2219
do_call at /buildworker/worker/package_linux64/build/src/interpreter.c:323
eval_value at /buildworker/worker/package_linux64/build/src/interpreter.c:411
eval_stmt_value at /buildworker/worker/package_linux64/build/src/interpreter.c:362 [inlined]
eval_body at /buildworker/worker/package_linux64/build/src/interpreter.c:773
jl_interpret_toplevel_thunk_callback at /buildworker/worker/package_linux64/build/src/interpreter.c:885
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7f258b31621f)
unknown function (ip: 0x6)
jl_interpret_toplevel_thunk at /buildworker/worker/package_linux64/build/src/interpreter.c:894
jl_toplevel_eval_flex at /buildworker/worker/package_linux64/build/src/toplevel.c:764
jl_toplevel_eval_in at /buildworker/worker/package_linux64/build/src/toplevel.c:793
eval at ./boot.jl:328
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2219
eval_user_input at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/REPL/src/REPL.jl:85
run_backend at /home/USERNAME/.julia/packages/Revise/yp5KG/src/Revise.jl:769
#58 at ./task.jl:259
jl_fptr_trampoline at /buildworker/worker/package_linux64/build/src/gf.c:1864
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2219
jl_apply at /buildworker/worker/package_linux64/build/src/julia.h:1571 [inlined]
start_task at /buildworker/worker/package_linux64/build/src/task.c:572
unknown function (ip: 0xffffffffffffffff)
Allocations: 61223335 (Pool: 61207664; Big: 15671); GC: 123
[1]    26169 segmentation fault (core dumped)  ~/Apps/julia-1.1.0/bin/julia 
```
I don't really follow the stacktrace, but if it can help…

For reference:
```julia
julia> versioninfo()
Julia Version 1.1.0
Commit 80516ca202 (2019-01-21 21:24 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-7600U CPU @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)
```

**Comment by quinnj** [comment]
Both of these examples are fixed in #395

**Comment by quinnj** [comment]
Fixed in #395
